### ISSUE TITLE
Unable to run 50 containers - rpc error: code = 2 desc = "oci runtime error..."

### ISSUE THREAD

**Issue by aanm** [docker/docker#24536]
**Output of `docker version`:**

```
$ docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:39:21 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:39:21 2016
 OS/Arch:      linux/amd64
```


**Output of `docker info`:**

```                                                                                                                 
[aanm@aanm-MBP ~]$ docker info                                                                                                                                                                                                               
Containers: 47                                                                                                                                                                                                                               
 Running: 45                                                                                                                                                                                                                                 
 Paused: 0                                                                                                                                                                                                                                   
 Stopped: 2                                                                                                                                                                                                                                  
Images: 2                                                                                                                                                                                                                                    
Server Version: 1.11.2                                                                                                                                                                                                                       
Storage Driver: overlay                                                                                                                                                                                                                      
 Backing Filesystem: extfs                                                                                                                                                                                                                   
Logging Driver: json-file                                                                                                                                                                                                                    
Cgroup Driver: cgroupfs                                                                                                                                                                                                                      
Plugins:                                                                                                                                                                                                                                     
 Volume: local                                                                                                                                                                                                                               
 Network: bridge null host                                                                                                                                                                                                                   
Kernel Version: 4.6.3-300.fc24.x86_64                                                                                                                                                                                                        
Operating System: Fedora 24 (Twenty Four)                                                                                                                                                                                                    
OSType: linux                                                                                                                                                                                                                                
Architecture: x86_64                                                                                                                                                                                                                         
CPUs: 8                                                                                                                                                                                                                                      
Total Memory: 15.58 GiB                                                                                                                                                                                                                      
Name: aanm-MBP
ID: L3I2:XPAF:HYX3:TFTI:RVWP:USWP:2ZTD:PA6M:22DR:UFUA:I7QM:GMMW
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Username: aanm
Registry: https://index.docker.io/v1/
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**



**Steps to reproduce the issue:**
1. Run 40ish containers
2. Run some more container until you see the issue
3.


**Describe the results you received:**
```
$ docker run -d ubuntu:16.04 sleep 3000000s
2b3ec5d5cb585e673cdbacb5e47c240d451ede37d5e0a0662e44c21f196a1a79                                                                                                                                                                             
docker: Error response from daemon: rpc error: code = 2 desc = "oci runtime error: read parent: connection reset by peer".  
```

**Describe the results you expected:**
No error messages

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by thaJeztah** [comment]
could you check if this relates to https://github.com/docker/docker/issues/22690#issuecomment-218757471 ?

**Comment by aanm** [comment]
@thaJeztah That solves my problem. May I suggest, if it doesn't exist yet, a Q&A with this small stuff? :-) You can close the issue

**Comment by thaJeztah** [comment]
Yeah, that setting is a bit of a PITA; newer versions of systemd apply the (way too low) tasks limit, so had to override that setting https://github.com/docker/docker/pull/19391. Unfortunately, older versions of systemd fail to start if that setting is set,  so we had to remove it https://github.com/docker/docker/pull/20167, and add it commented-out to give a clue https://github.com/docker/docker/pull/21491. Because not everyone will read that file (or reads docs), we tried to set it automatically, depending on the systemd version, but that cause the version number to be messed up https://github.com/docker/docker/pull/21628, had to revert https://github.com/docker/docker/pull/24100, and finally hope we have all issues solved to set it automatically for docker 1.12 :smile: https://github.com/docker/docker/pull/24558 :dancer: 

I'll close this issue, but feel free to comment after I closed :smile:
